core/src/docs/upgrade.md

```diff
@@ -11,6 +11,19 @@ Users will need to change the following usage:
 
 `delimiter` other than `""` and `"/"` is not supported anymore.
 
+## Raw API
+
+### Lister Align
+
+We changed our internal `lister` implementation to align with the `list` public API for better performance and readability.
+
+- trait `Page` => `List`
+- struct `Pager` => `Lister`
+- trait `BlockingPage` => `BlockingList`
+- struct `BlockingPager` => `BlockingLister`
+
+Every call to `next` will return an entry instead a page of entries. Also, we changed our async list api into poll based instead of `async_trait`.
```

Comment:
```rust
#[async_trait]
impl<R: oio::List> oio::List for ConcurrentLimitWrapper<R> {
    fn poll_next(&mut self, cx: &mut Context<'_>) -> Poll<Result<Option<oio::Entry>>> {
        self.inner.poll_next(cx)
    }
}
```
As you said, using the code I posted as an example, I think we can remove `#[async_trait]` now.
I tried it locally, and it works.

Is this thought right?

Reply:
Yes, we can remove them. Please allow me to do it in another PR so that we don't block here.

Reply:
SGTM.

Reply:
Tracked at https://github.com/apache/incubator-opendal/issues/3601